custom/enikshay/management/commands/generate_nikshay_notification_report.py

```diff
@@ -0,0 +1,214 @@
+from datetime import datetime
+from openpyxl import Workbook
+import csv
+
+from django.core.management.base import BaseCommand
+from corehq.form_processor.exceptions import CaseNotFound
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.motech.repeaters.dbaccessors import get_repeat_records_by_payload_id
+from corehq.pillows.mappings.case_search_mapping import CASE_SEARCH_MAX_RESULTS
+from corehq.apps.es.case_search import CaseSearchES
+from corehq.util.log import with_progress_bar
+
+from casexml.apps.case.xml.parser import CaseUpdateAction
+from casexml.apps.case.xform import get_case_updates
+from custom.enikshay.case_utils import get_person_case_from_episode, get_open_episode_case_from_person
+from custom.enikshay.const import DSTB_EPISODE_TYPE
+from custom.enikshay.exceptions import ENikshayCaseNotFound
+from custom.enikshay.integrations.utils import is_valid_person_submission
+
+DOMAIN = "enikshay"
+
+
+class Command(BaseCommand):
+    help = """
+    Iterate over a list of episode case ids and check if they have been
+    notified to nikshay or not
+    Pass in a csv file that has
+    1. episode case ids under header episode_id or
+    2. person enikshay ids(not uuid) under header enikshay_id (along with option --parse_as_person_enikshay_ids)
+
+    Gives a report with:
+    'Enikshay ID', 'Episode ID', 'Should be Notified', 'Nikshay Registered', 'Nikshay Private Registered',
+    'Nikshay ID', 'External ID', 'Nikshay ID history', 'Nikshay Error', 'Private Nikshay Error',
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.person_enikshay_ids = []
+        self.person_ids = []
+        self.episode_ids = []
+        self.max_attempts_in_sheet = 0
+        self.search = CaseSearchES().domain('enikshay').size(CASE_SEARCH_MAX_RESULTS)
+        self.wb = Workbook()
+        self.ws = self.wb.active
+        self.ws.append([
+            'Enikshay ID', 'Episode ID', 'Should be Notified', 'Nikshay Registered', 'Nikshay Private Registered',
+            'Nikshay ID', 'External ID', 'Nikshay ID history', 'Nikshay Error', 'Private Nikshay Error',
+            # Forwarding ID 1, Attempts 1, Forwarding ID 2, Attempts 2 ..
+        ])
+
+    def add_arguments(self, parser):
+        parser.add_argument('file_path')
+        parser.add_argument(
+            '--parse_as_person_enikshay_ids',
+            dest="parse_as_person_enikshay_ids",
+            action='store_true',
+            help='pass this to let parser know to look for enikshay ids instead of episode ids in file'
+        )
+
+    def _search_person_using_enikshay_id(self, enikshay_id):
+        search_results = self.search.case_property_query("person_id", enikshay_id).run()
+        return [case['_id'] for case in search_results.raw_hits]
```

Comment:
you can use `self.search.case_property_query("person_id", enikshay_id).values_list(['_id', ], flat=True)` instead of this.